drivers/dac/ad3552r/ad3552r.c

```diff
@@ -607,16 +605,44 @@ static int32_t _ad3552r_set_code_value(struct ad3552r_desc *desc,
 				       uint8_t ch,
 				       uint16_t val)
 {
-	uint16_t code;
 	uint8_t addr;
 
-	addr = _get_code_reg_addr(ch, 1, 0);
+	addr = ad3552r_get_code_reg_addr(ch, 1, desc->ch_data[ch].fast_en);
+	return ad3552r_write_reg(desc, addr, val);
+}
+
+int32_t ad3552r_simulatneous_update_enable(struct ad3552r_desc *desc)
+{
+	if (desc->ch_data[0].fast_en != desc->ch_data[1].fast_en)
+		return -EINVAL;
+	if (desc->is_simultaneous) {
+		if (desc->ch_data[0].fast_en)
```

Comment:
some more code optimization that can be done in your patch: if you have the following scenario:
```
if (statement)
   return x;
else
   return y;
```
This can be reduced to:
```
if (statement)
   return x;
return y;
```
There are a lot of occurrences like this in the code that can be optimized.

Reply:
Addressed this comment.